pallets/asset-manager/src/lib.rs

```diff
@@ -296,15 +306,11 @@ pub mod pallet {
     pub(super) type LocationAssetId<T: Config> =
         StorageMap<_, Blake2_128Concat, T::Location, T::AssetId>;
 
-    /// AssetId to AssetRegistry Map.
+    /// AssetId to AssetMetadata Map.
     #[pallet::storage]
     #[pallet::getter(fn asset_id_metadata)]
-    pub(super) type AssetIdMetadata<T: Config> = StorageMap<
-        _,
-        Blake2_128Concat,
-        T::AssetId,
-        <T::AssetConfig as AssetConfig<T>>::AssetRegistryMetadata,
-    >;
+    pub(super) type AssetIdMetadata<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AssetId, AssetMetadata<T::Balance>>;
```

Comment:
Do we need migration here? Not sure any change in this metadata type.

Reply:
Yes we do, this PR turns the storage item into the enum `AssetMetadata`, Storage is currently `FungibleAssetRegistryMetadata` on calamari.

Reply:
same as https://github.com/Manta-Network/Manta/pull/1012#discussion_r1097409189